api/src/server.rs

```diff
@@ -876,130 +878,47 @@ where
 
     async fn get_stream_state(
         &self,
-        pipeline: &SessionContext,
+        aggregator: AggregatorHandle,
         stream_id: StreamId,
     ) -> Result<ceramic_api_server::StreamsStreamIdGetResponse, ErrorResponse> {
-        let state_batch = pipeline
-            .table(EVENT_STATES_TABLE)
-            .await
-            .map_err(|err| {
-                ErrorResponse::new(format!("{} table not found: {err}", EVENT_STATES_TABLE))
-            })?
-            .select(vec![
-                col("stream_cid"),
-                col("event_cid"),
-                col("dimensions"),
-                col("controller"),
-                col("data"),
-                col("index"),
-            ])
-            .map_err(|err| ErrorResponse::new(format!("failed to select: {err}")))?
-            .filter(col("stream_cid").eq(lit(stream_id.cid.to_bytes())))
-            .map_err(|err| ErrorResponse::new(format!("failed to filter: {err}")))?
-            .aggregate(
-                vec![col("stream_cid"), col("controller")],
-                vec![
-                    last_value(vec![col("data")])
-                        .order_by(vec![col("index").sort(true, true)])
-                        .build()
-                        .map_err(|err| {
-                            ErrorResponse::new(format!(
-                                "failed to define last_value state query: {err}"
-                            ))
-                        })?
-                        .alias("data"),
-                    last_value(vec![col("event_cid")])
-                        .order_by(vec![col("index").sort(true, true)])
-                        .build()
-                        .map_err(|err| {
-                            ErrorResponse::new(format!(
-                                "failed to define last_value event_cid query: {err}"
-                            ))
-                        })?
-                        .alias("event_cid"),
-                    last_value(vec![col("dimensions")])
-                        .order_by(vec![col("index").sort(true, true)])
-                        .build()
-                        .map_err(|err| {
-                            ErrorResponse::new(format!(
-                                "failed to define last_value event_cid dimensions: {err}"
-                            ))
-                        })?
-                        .alias("dimensions"),
-                ],
-            )
-            .map_err(|err| ErrorResponse::new(format!("failed to define window: {err}")))?
-            .collect()
+        let state = aggregator
```

Comment:
And here is how our own code can consume the actors. This is not subscribing to the actor, just making a stream state query.

Reply:
Yeah, this make sense (and is nicer than before), which means we can write a query against the flight sql endpoint that will wait until the data exists. I guess I'm wondering about SSE/WS updates. Could we provide a new route that spawns a task for the connection and uses `SubscribeSinceMsg` to receive updates and then push them out or am I thinking about this wrong?

Reply:
The plan is to use FlightSQL native streaming support instead of SSE or WS. I have a WIP PR I hope to get up today/tomorrow that adds support for these queries to the SDK. 

We could add SSE/WS support if there is enough demand for it, but I expect/hope that FlightSQL is sufficient.

Reply:
I pushed up what I have here https://github.com/ceramicnetwork/ceramic-sdk/pull/48

Reply:
nice, that makes sense